src/Common/tests/System/Diagnostics/DebuggerAttributes.cs

```diff
@@ -67,65 +69,112 @@ private static Type GetProxyType(Type type, Type[] genericTypeArguments)
             return proxyType;
         }
 
-        internal static void ValidateDebuggerDisplayReferences(object obj)
+        internal static string ValidateDebuggerDisplayReferences(object obj)
```

Comment:
I think it should be possible to change the API for validating type proxies to actually return the type proxy, too. I'll have to think about that later though since I'm a little busy today.